packages/adminrouter/extra/src/test-harness/tests/test_metrics.py

```diff
@@ -95,3 +96,51 @@ def test_metrics_prometheus_escape(self, master_ar_process, valid_user_header):
         assert '/service/monitoring/gra"f\\a\nn\ta' not in resp.text
         # correctly escaped:
         assert '/service/monitoring/gra\\"f\\\\a\\nn\ta' in resp.text
+
+    def test_metrics_prometheus_histogram(self, master_ar_process, mocker, valid_user_header):
+        """
+        Response times are measured in histogram output.
+        """
+        mocker.send_command(endpoint_id='http://127.0.0.1:8181',
+                            func_name='always_stall',
+                            aux_data=0.05)
+
+        url = master_ar_process.make_url_from_path('/exhibitor/')
+
+        resp = requests.get(
+            url,
+            allow_redirects=False,
+            headers=valid_user_header)
+
+        mocker.send_command(endpoint_id='http://127.0.0.1:8181',
+                            func_name='always_stall',
+                            aux_data=0.3)
+
+        resp = requests.get(
+            url,
+            allow_redirects=False,
+            headers=valid_user_header)
+
+        url = master_ar_process.make_url_from_path('/nginx/metrics')
+
+        resp = requests.get(
+            url,
+            allow_redirects=False
+        )
+
+        assert resp.status_code == 200
+        assert resp.headers['Content-Type'] == 'text/plain'
+        prefix = (
+            'nginx_vts_filter_request_duration_seconds_bucket{'
+            'filter="upstream:=Exhibitor",'
+            'filter_name="backend:=127.0.0.1:8181"'
```

Comment:
why do I need both, a filter and a filter name? Logically, one of seems to be sufficient to filter for Exhibitor :-)

Reply:
clarified in person. This is not part of the request, but just double-checking that both are present in the response